toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1485,16 +1485,32 @@ func ProvisionUserSSHCerts(installChroot safechroot.ChrootInterface, username st
 	const squashErrors = false
 	const authorizedKeysTempFilePerms = 0644
 	const authorizedKeysTempFile = "/tmp/authorized_keys"
-	const sshDirectoryPermission = "0700"
+	const sshDirectoryPermission = 0700
 
 	// Skip user SSH directory generation when not provided with public keys
 	// Let SSH handle the creation of this folder on its first use
-	if len(sshPubKeyPaths) == 0 {
+	if len(sshPubKeyPaths) == 0 && len(sshPubKeys) == 0 {
 		return
 	}
 
 	homeDir := userutils.UserHomeDirectory(username)
 	userSSHKeyDir := filepath.Join(homeDir, ".ssh")
+
+	exists, err = file.PathExists(userSSHKeyDir)
+	if err != nil {
+		logger.Log.Warnf("Error accessing %s directory: %v", userSSHKeyDir, err)
+		return
+	}
+	if !exists {
+		logger.Log.Debugf("Directory %s does not exist. Creating directory...", userSSHKeyDir)
+
+		err = os.Mkdir(userSSHKeyDir, os.FileMode(sshDirectoryPermission))
+		if err != nil {
+			logger.Log.Warnf("Failed to create %s directory: %v", userSSHKeyDir, err)
+			return
+		}
+	}
+
```

Comment:
I believe the `err = installChroot.AddFiles(fileToCopy)` line below is what creates the .ssh directory. So, I don’t think this change is necessary.

(GitHub doesn’t allow you to add comments to lines that are not a part of the PR. )

Reply:
you're right. I will revert it and update this pr to only include the bug in the if condition